test/src/test_swap_endian.cpp

```diff
@@ -143,8 +177,57 @@ TEST(EndianSwapUtility, TestSwapEndianNestedFloatAndDoubleArray)
   auto swapped_nested_float_and_double_array_struct =
       endian_swap_utility::swap_endian(nested_float_and_double_array_struct);
 
-  EXPECT_EQ(swapped_nested_float_and_double_array_struct.a[0], little_to_big_endian(1.0f));
-  EXPECT_EQ(swapped_nested_float_and_double_array_struct.a[1], little_to_big_endian(2.0f));
-  EXPECT_EQ(swapped_nested_float_and_double_array_struct.b[0], little_to_big_endian(3.0));
-  EXPECT_EQ(swapped_nested_float_and_double_array_struct.b[1], little_to_big_endian(4.0));
-}
\ No newline at end of file
+  EXPECT_EQ(swapped_nested_float_and_double_array_struct.a[0], endian_swap(1.0f));
+  EXPECT_EQ(swapped_nested_float_and_double_array_struct.a[1], endian_swap(2.0f));
+  EXPECT_EQ(swapped_nested_float_and_double_array_struct.b[0], endian_swap(3.0));
+  EXPECT_EQ(swapped_nested_float_and_double_array_struct.b[1], endian_swap(4.0));
+}
+
+TEST(EndianSwapUtility, TestPodClass)
+{
+  
+  EXPECT_TRUE(std::is_pod<Podclass>::value);
+  EXPECT_FALSE(std::is_pod<NonPodclass>::value);
+
+  Podclass pc{1, 2.0f, 3.0};
+  ASSERT_EQ(sizeof(pc), sizeof(int) + sizeof(float) + sizeof(double));
+  ASSERT_EQ(sizeof(float), 4);
+
+  auto endianswap = endian_swap_utility::swap_endian(pc);
+  EXPECT_EQ(endianswap.b, endian_swap(2.0f));
+  EXPECT_EQ(endianswap.c, endian_swap(3.0));
+  EXPECT_EQ(0x12345678, __bswap_32(0x78563412));
+  EXPECT_EQ(0x1234567890abcdef, __bswap_64(0xefcdab9078563412));
+
+  EXPECT_EQ(endianswap.a, __bswap_32(pc.a));
+
+  auto sw_b =  __bswap_32(*reinterpret_cast<uint32_t *>(&pc.b));
+  EXPECT_EQ(endianswap.b, *reinterpret_cast<float *>(&sw_b));
+
+  auto sw_c = __bswap_64(*reinterpret_cast<uint64_t *>(&pc.c));
+  EXPECT_EQ(endianswap.c, *reinterpret_cast<double *>(&sw_c));
+
+}
+
+TEST(EndianSwapUtility, TestPod)
+{
+  double a = 3.1415926;
+  auto endianswap = endian_swap_utility::swap_endian(a);
+  auto sw_a = __bswap_64(*reinterpret_cast<uint64_t *>(&a));
```

Comment:
`__bswap_64` is specialized on x86_64, not portable, should we consider using posix interfaces ?
```suggestion
  auto sw_a = __bswap_64(*reinterpret_cast<uint64_t *>(&a));
```

Reply:
ok, change to  a portable style of swap.